### PR TITLE
Add support for patch command

### PR DESCRIPTION
**Pull request by bradbeam**
Let's try this again. 

This should be a bit more concise and probably closer in line with expectations.

Example usage
```shell 
bradbeam@cr-2:~/go/src/github.com/redskyops/redskyops-controller/redskyctl
[22:36:01] # kustomize build $recipes/postgres | ./redskyctl patch --file - --trialnumber 24 | grep -e cpu -e mem
            cpu: 615m
            memory: 500Mi
            cpu: 615m
            memory: 500Mi
bradbeam@cr-2:~/go/src/github.com/redskyops/redskyops-controller/redskyctl
[22:36:13] # kustomize build $recipes/postgres | ./redskyctl patch --file - --trialnumber 25 | grep -e cpu -e mem
            cpu: 585m
            memory: 1275Mi
            cpu: 585m
            memory: 1275Mi
bradbeam@cr-2:~/go/src/github.com/redskyops/redskyops-controller/redskyctl
[22:36:17] # ./redskyctl patch --file $recipes/postgres/experiment.yaml --file $recipes/postgres/application/postgres.yaml --trialnumber 24 | grep -e cpu -e mem
            cpu: 615m
            memory: 500Mi
            cpu: 615m
            memory: 500Mi
bradbeam@cr-2:~/go/src/github.com/redskyops/redskyops-controller/redskyctl
[22:36:34] # ./redskyctl patch --file $recipes/postgres/experiment.yaml --file $recipes/postgres/application/postgres.yaml --trialnumber 25 | grep -e cpu -e mem
            cpu: 585m
            memory: 1275Mi
            cpu: 585m
            memory: 1275Mi
```

replaces #110 #120 

Signed-off-by: Brad Beam <brad.beam@carbonrelay.com>

### CODE REVIEW

**Comment by jgustie** [comment]
Instead of `--trialnumber` can we take a single argument so this is more consistent with `redskyctl get`?

File: redskyctl/internal/commands/export/export.go [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+		if err := o.Fs.WriteFile(assetName, listBytes); err != nil {
+			return err
+		}
+
+		o.resources[assetName] = struct{}{}
+
+		if te, ok := list.Items[idx].Object.(*redsky.Experiment); ok {
+			o.experiment = &redsky.Experiment{}
+			te.DeepCopyInto(o.experiment)
+			break
```
Comment (bradbeam) [inline redskyctl/internal/commands/export/export.go]: dont break here

File: redskyctl/internal/commands/export/export.go [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintln(o.Out, string(yamls))
+
+	return nil
+}
+
+func (o *Options) generateExperiment() error {
```
Comment (bradbeam) [inline redskyctl/internal/commands/export/export.go:357]: add `app.filterByExperiment(o.trialName(minus id))` to reduce the application down to only the stuff we care about

File: redskyctl/internal/commands/export/export.go [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+	gen.ContainerResourcesSelectors = experimentctl.DefaultContainerResourcesSelectors()
+
+	if gen.Application.Parameters != nil && gen.Application.Parameters.ContainerResources != nil {
+		ls := labels.Set(gen.Application.Parameters.ContainerResources.Labels).String()
+
+		for i := range gen.ContainerResourcesSelectors {
+			gen.ContainerResourcesSelectors[i].LabelSelector = ls
+		}
+	}
+
```
Comment (bradbeam) [inline redskyctl/internal/commands/export/export.go]: this should go away

**Pull request closed without merging**